core/src/labels/labels.cpp

```diff
@@ -108,6 +109,65 @@ void Labels::update(const View& _view, float _dt, const std::vector<std::unique_
     }
 }
 
+const std::vector<std::shared_ptr<Properties>>& Labels::getFeaturesAtPoint(const View& _view, float _dt,
+                                                                           const std::vector<std::unique_ptr<Style>>& _styles,
+                                                                           const std::vector<std::shared_ptr<Tile>>& _tiles,
+                                                                           float _x, float _y, bool _visibleOnly) {
+    // FIXME dpi dependent threshold
+    const float thumbSize = 50;
+
+    m_touchItems.clear();
+
+    glm::vec2 screenSize = glm::vec2(_view.getWidth(), _view.getHeight());
+    glm::vec2 touchPoint(_x, _y);
+
+    OBB obb(_x - thumbSize/2, _y - thumbSize/2, 0, thumbSize, thumbSize);
+
+    std::shared_ptr<Properties> selectedItem;
+    float minDistance = std::numeric_limits<float>::max();
+
+    for (const auto& tile : _tiles) {
+
+        glm::mat4 mvp = _view.getViewProjectionMatrix() * tile->getModelMatrix();
+
+        for (const auto& style : _styles) {
+            const auto& mesh = tile->getMesh(*style);
+            if (!mesh) { continue; }
+
+            const LabelMesh* labelMesh = dynamic_cast<const LabelMesh*>(mesh.get());
+            if (!labelMesh) { continue; }
+
+            for (auto& label : labelMesh->getLabels()) {
+
+                auto& options = label->getOptions();
+                if (!options.interactive) { continue; }
+
+                if (!_visibleOnly) {
+                    label->updateScreenTransform(mvp, screenSize, false);
+                    label->updateBBoxes();
+                }
+
+                if (isect2d::intersect(label->getOBB(), obb)) {
+                    float distance = glm::length2(label->getTransform().state.screenPos - touchPoint);
+
+                    if (distance < minDistance) {
+                        minDistance = distance;
+                        selectedItem = options.properties;
+                    }
+                }
+            }
+        }
+    }
+
+    if (selectedItem) {
```

Comment:
Should this not be within the for loop?

Else, `selectedItem` will be overwritten by previous and `m_touchItems` will always contain the last `selectedItem`.